### PR TITLE
Make constant_pad_nd ref decomposition fully functional

### PR DESCRIPTION
**Pull request by angelayi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #185636

The constant_pad_nd ref decomposition used prims.copy_to to copy input
data into a narrowed view of the output tensor. prims.copy_to is a
mutable primitive that has no functionalization kernel: it has no
CompositeImplicitAutograd dispatch, no generated C++ Functionalize
kernel (codegen only covers aten ops), and auto_functionalize rejects
it as a builtin op. This means it survives functionalization as a dead
node (num_users=0), and then trips assert_functional_graph in
aot_autograd.

This manifests when torch.onnx.export(dynamo=True) is used on models
containing F.pad (e.g., HuggingFace GraniteMoeHybrid's Mamba layer
pads hidden states to chunk_size boundaries). The ONNX export path's
run_decompositions step uses a decomp table that includes
constant_pad_nd, triggering the prims.copy_to decomposition.

Replace the mutable empty+fill+narrow+copy_to pattern with a fully
functional torch.cat approach: for each padded dimension, concatenate
torch.full padding tensors around the input. This decomposition is
only in decomposition_table (not used by torch.compile or
core_aten_decompositions), so the extra intermediates have no
performance impact on any compile path.

Fixes https://github.com/pytorch/pytorch/issues/167068

Test Plan:
Verified the fix resolves the original issue with a minimal repro
of the Mamba pad+chunk pattern through the ONNX export path:

```
python -m pytest test/test_ops.py -x -k "constant_pad_nd"
# 144 passed, 22 skipped

python -m pytest test/test_nn.py -x -k "constant_pad"
# 1 passed
```

Added regression test in test/export/test_export.py that exercises
F.pad through export + run_decompositions with decomposition_table.

### CODE REVIEW

**Comment by angelayi** [comment]
@claude review

**Review by justinchuby — APPROVED** [review]
Thank you!

**Review by zou3519 — APPROVED** [review]
